File: AES.py
from Crypto.Cipher import AES
import os
import base64
from itsdangerous import base64_encode, base64_decode

BS = AES.block_size
print(BS)
pad = lambda s: s + bytes([BS - len(s) % BS]) * (BS - len(s) % BS)
unpad = lambda s: s[0:-ord("%c" % s[-1])]

key = b'1234567890123456'
iv = b'1234567890123456'

text_lt = b'1234567890'
text_eq = b'1234567890abcdef'
text_gt = b'1234567890abcdef123456'

cipher = AES.new(key, AES.MODE_CBC, IV=iv)
data_lt = base64.b64encode(cipher.encrypt(pad(text_lt)))
print("原字符串是: %s" % str(text_lt, encoding = "utf-8"))
print("加密后的字串是: %s" % str(data_lt, encoding = "utf-8"))
cipher = AES.new(key, AES.MODE_CBC, IV=iv)
print("解密后的字串是: %s" % str(unpad(cipher.decrypt(base64_decode(data_lt))), encoding='utf-8'))

cipher = AES.new(key, AES.MODE_CBC, IV=iv)
data_eq = base64.b64encode(cipher.encrypt(pad(text_eq)))
print("原字符串是: %s" % str(text_eq, encoding = "utf-8"))
print("加密后的字串是: %s" % str(data_eq, encoding = "utf-8"))
cipher = AES.new(key, AES.MODE_CBC, IV=iv)
print("解密后的字串是: %s" % str(unpad(cipher.decrypt(base64_decode(data_eq))), encoding='utf-8'))


cipher = AES.new(key, AES.MODE_CBC, IV=iv)
data_gt = base64.b64encode(cipher.encrypt(pad(text_gt)))
print("原字符串是: %s" % str(text_gt, encoding = "utf-8"))
print("加密后的字串是: %s" % str(data_gt, encoding = "utf-8"))
cipher = AES.new(key, AES.MODE_CBC, IV=iv)
print("解密后的字串是: %s" % str(unpad(cipher.decrypt(base64_decode(data_gt))), encoding='utf-8'))

File: AES.java
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

import sun.misc.BASE64Decoder;
import sun.misc.BASE64Encoder;

public class AES {

    // 加密
    public static String Encrypt(String sSrc, String sKey) throws Exception {
        if (sKey == null) {
            System.out.print("Key为空null");
            return null;
        }
        // 判断Key是否为16位
        if (sKey.length() != 16) {
            System.out.print("Key长度不是16位");
            return null;
        }
        byte[] raw = sKey.getBytes();
        SecretKeySpec skeySpec = new SecretKeySpec(raw, "AES");
        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");//"算法/模式/补码方式"
        IvParameterSpec iv = new IvParameterSpec("1234567890123456".getBytes());//使用CBC模式，需要一个向量iv，可增加加密算法的强度
        cipher.init(Cipher.ENCRYPT_MODE, skeySpec, iv);
        byte[] encrypted = cipher.doFinal(sSrc.getBytes());

        return new BASE64Encoder().encode(encrypted);//此处使用BASE64做转码功能，同时能起到2次加密的作用。
    }

    // 解密
    public static String Decrypt(String sSrc, String sKey) throws Exception {
        try {
            // 判断Key是否正确
            if (sKey == null) {
                System.out.print("Key为空null");
                return null;
            }
            // 判断Key是否为16位
            if (sKey.length() != 16) {
                System.out.print("Key长度不是16位");
                return null;
            }
            byte[] raw = sKey.getBytes("ASCII");
            SecretKeySpec skeySpec = new SecretKeySpec(raw, "AES");
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            IvParameterSpec iv = new IvParameterSpec("1234567890123456"
                    .getBytes());
            cipher.init(Cipher.DECRYPT_MODE, skeySpec, iv);
            byte[] encrypted1 = new BASE64Decoder().decodeBuffer(sSrc);//先用base64解密
            try {
                byte[] original = cipher.doFinal(encrypted1);
                String originalString = new String(original);
                return originalString;
            } catch (Exception e) {
                System.out.println(e.toString());
                return null;
            }
        } catch (Exception ex) {
            System.out.println(ex.toString());
            return null;
        }
    }

    public static void main(String[] args) throws Exception {
        String cKey = "1234567890123456";
        String cSrc = "1234567890";
        System.out.println("原字符串是:" + cSrc);
        String enString = AES.Encrypt(cSrc, cKey);
        System.out.println("加密后的字串是：" + enString);
        String DeString = AES.Decrypt(enString, cKey);
        System.out.println("解密后的字串是：" + DeString);

        cSrc = "1234567890abcdef";
        System.out.println("原字符串是:" + cSrc);
        enString = AES.Encrypt(cSrc, cKey);
        System.out.println("加密后的字串是：" + enString);
        DeString = AES.Decrypt(enString, cKey);
        System.out.println("解密后的字串是：" + DeString);

        cSrc = "1234567890abcdef123456";
        System.out.println("原字符串是:" + cSrc);
        enString = AES.Encrypt(cSrc, cKey);
        System.out.println("加密后的字串是：" + enString);
        DeString = AES.Decrypt(enString, cKey);
        System.out.println("解密后的字串是：" + DeString);

    }
}